src/iuse.cpp

```diff
@@ -6029,6 +6030,18 @@ int iuse::vacutainer(player *p, item *it, bool, point)
 
     if (!drew_blood && query_yn(_("Draw your own blood?"))) {
         drew_blood = true;
+        if (p->has_trait ("ACIDBLOOD")) {
+            it->put_in(acid);
+            if (one_in(2) && it->damage <= 3){
+                it->damage++;
+                p->add_msg_if_player(m_info, _("Your acidic blood damages the %s!"), it->tname().c_str());
+            }
+            if (!one_in(4) && it->damage >= 4){
+                p->inv.remove_item(it);
```

Comment:
This will delete the item `it`, you can not refer to it anymore after that. The message should be printed before removing it, also the function should be left directly with `return 0;`, not referring to the non-existing item anymore (and reducing charges from it doesn't make any sense as it's already gone anyway).

Btw. this clearly belongs into the Xenomorph mutation branch.